src/TestFramework/MSTest.Core/Resources/FrameworkMessages.resx

```diff
@@ -292,4 +292,11 @@ Stack Trace: {4}</value>
   <data name="DynamicDataIEnumerableEmpty" xml:space="preserve">
     <value>Property or method {0} on {1} returns empty IEnumerable&lt;object[]&gt;.</value>
   </data>
+  <data name="ArgumentXMustDeriveFromClassY" xml:space="preserve">
+    <value>{0} must derive from {1}!</value>
```

Comment:
What about

```suggestion
    <value>Type '{0}' is not assignable to '{1}'.</value>
```

Reply:
And maybe I would make the resource entry less generic and directly hardcode the `{1}` (and rename the resource entry into `TypeIsNotAssignableToApplication`).